genericapi/handler.go

```diff
@@ -158,7 +158,7 @@ func (h *Handler) ServeCreateAlert(w http.ResponseWriter, r *http.Request) {
 	}
 
 	err = retry.DoTemporaryError(func(int) error {
-		createdAlert, isNew, err := h.c.AlertStore.CreateOrUpdate(ctx, a, meta)
+		createdAlert, isNew, err := h.c.AlertStore.CreateOrUpdate(ctx, a)
```

Comment:
I understand that we're handling it with a specific store method, But I couldn't to find any references to the store method `SetMetadataTx` for generic api call.
Is the plan is to support adding meta only via graphql?

Reply:
That's not a bad idea. The method to set metadata is in `alert/metadata.go` in the latest commit.

What you have here looks good, two thoughts:
- Have the `CreateOrUpdate` method just call the new *Meta version with `nil`
- Only set metadata if `isNew` is true so we can treat it as write-once (i.e., only on alert creation) at least to start out

Reply:
Do you mean just the update the current `CreateOrUpdate` method to have new arg *Meta? In that case, I'll have to update all the other clients(grafana, prometheus, ...) to update it to have nil values, Is that okay?